sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharpSensor.java

```diff
@@ -80,22 +79,25 @@ private static boolean shouldExecuteOnProject(FileSystem fs) {
   }
 
   private void executeInternal(SensorContext context) {
-    Optional<Path> roslynReportPath = config.roslynReportPath();
-    Optional<Path> protobufReportsDirectory = config.protobufReportPath();
+    List<Path> protobufPaths = reportPathCollector.protobufDirs();
+    List<Path> roslynDirs = reportPathCollector.roslynDirs();
 
-    if (protobufReportsDirectory.isPresent()) {
-      LOG.info("Importing analysis results from " + protobufReportsDirectory.get().toAbsolutePath());
-      protobufDataImporter.importResults(context, protobufReportsDirectory.get(), CSharpSonarRulesDefinition.REPOSITORY_KEY, !roslynReportPath.isPresent());
+    if (!protobufPaths.isEmpty()) {
+      protobufDataImporter.importResults(context, protobufPaths, CSharpSonarRulesDefinition.REPOSITORY_KEY, roslynDirs.isEmpty());
     }
 
-    if (roslynReportPath.isPresent()) {
+    if (!roslynDirs.isEmpty()) {
       LOG.info("Importing Roslyn report");
       Map<String, List<RuleKey>> activeRoslynRulesByPartialRepoKey = RoslynProfileExporter.activeRoslynRulesByPartialRepoKey(context.activeRules()
         .findAll()
         .stream()
         .map(ActiveRule::ruleKey)
         .collect(toList()));
-      roslynDataImporter.importRoslynReport(roslynReportPath.get(), context, activeRoslynRulesByPartialRepoKey);
+      roslynDataImporter.importRoslynReport(roslynDirs, context, activeRoslynRulesByPartialRepoKey);
     }
   }
+
+  private static String pluralize(String s, int count) {
+    return count == 1 ? s : s + "s";
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add parentheses to make the operator precedence explicit. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS864)